src/main/java/dumpling/Parser.java

```diff
@@ -0,0 +1,217 @@
+package dumpling;
+
+import dumpling.command.*;
+import dumpling.task.Deadline;
+import dumpling.task.Event;
+import dumpling.task.Task;
+import dumpling.task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Parser {
+
+    private static final String DEADLINE_DATE_SEPARATOR = "/by";
+    private static final String EVENT_FROM_SEPARATOR = "/from";
+    private static final String EVENT_TO_SEPARATOR = "/to";
+
+    /**
+     * Parse a given full command from the user and stores its information into the relevant Command object
+     *
+     * @param stringCommand user's input
+     * @return Command object that stores the information of the command. Resulting Command object can be executed.
+     * @throws DumplingException Message includes information about the error experienced.
+     */
+    public static Command parse(String stringCommand) throws DumplingException {
+        String commandString = stringCommand.split(" ")[0];
+        CommandEnum commandEnum;
+        try {
+            commandEnum = CommandEnum.valueOf(commandString.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new DumplingException(
+                    String.format(
+                            "%s is not a valid command!\n" +
+                                    "     To list items, use 'list'.\n" +
+                                    "     To mark or unmark an item as done, use '<mark/unmark> <item index>'.\n" +
+                                    "     To add a new item, use '<todo/deadline/event> <task name> <args>'.",
+                            stringCommand));
+        }
+        switch (commandEnum) {
+        case TODO:
```

Comment:
I like how you formatted your switch statements, together with your enums! Many would forget the //fall through comment.